File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/recyclertablayout/DemoImagePagerAdapter.java
package vn.loitp.app.activity.customviews.recyclerview.recyclertablayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.DrawableRes;
import androidx.viewpager.widget.PagerAdapter;

import java.util.ArrayList;
import java.util.List;

import loitp.basemaster.R;

public class DemoImagePagerAdapter extends PagerAdapter {

    private List<Integer> mItems = new ArrayList<>();

    public DemoImagePagerAdapter() {
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = LayoutInflater.from(container.getContext())
                .inflate(R.layout.layout_page, container, false);

        TextView textView = view.findViewById(R.id.title);
        textView.setText("Page: " + position);
        container.addView(view);

        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @DrawableRes
    public int getImageResourceId(int position) {
        return mItems.get(position);
    }

    public void addAll(List<Integer> items) {
        mItems = new ArrayList<>(items);
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/recyclerTabLayout/DemoImagePagerAdapter.kt
package vn.loitp.app.activity.customviews.recyclerview.recyclerTabLayout

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.DrawableRes
import androidx.viewpager.widget.PagerAdapter
import kotlinx.android.synthetic.main.view_item_recycler_tablayout_page.view.*
import vn.loitp.R

class DemoImagePagerAdapter : PagerAdapter() {

    private var mItems: List<Int> = ArrayList()

    @SuppressLint("SetTextI18n")
    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = LayoutInflater.from(container.context)
            .inflate(R.layout.view_item_recycler_tablayout_page, container, false)

        view.tvTitle.text = "Page: $position"
        container.addView(view)

        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, any: Any) {
        container.removeView(any as View)
    }

    override fun getCount(): Int {
        return mItems.size
    }

    override fun isViewFromObject(view: View, any: Any): Boolean {
        return view === any
    }

    @DrawableRes
    fun getImageResourceId(position: Int): Int {
        return mItems[position]
    }

    fun addAll(items: List<Int>) {
        mItems = ArrayList(items)
    }
}
